x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_initializer.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiFormRow,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FC } from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
+import { ENABLE_ESQL } from '@kbn/esql-utils';
+import type { AggregateQuery } from '@kbn/es-query';
+import { DataSourceTypeSelector } from './field_stats_initializer_view_type';
+import { useDataVisualizerKibana } from '../../../kibana_context';
+import { FieldStatsESQLEditor } from './field_stats_esql_editor';
+import type {
+  FieldStatisticsTableEmbeddableState,
+  FieldStatsInitialState,
+} from '../grid_embeddable/types';
+import { FieldStatsInitializerViewType } from '../grid_embeddable/types';
+import { isESQLQuery } from '../../search_strategy/requests/esql_utils';
+export interface FieldStatsInitializerProps {
+  initialInput?: Partial<FieldStatisticsTableEmbeddableState>;
+  onCreate: (props: FieldStatsInitialState) => void;
+  onCancel: () => void;
+  onPreview: (update: Partial<FieldStatsInitialState>) => Promise<void>;
+}
+
+const defaultESQLQuery = { esql: 'from kibana_sample_data_ecommerce | limit 10' };
+const defaultTitle = i18n.translate('xpack.dataVisualizer.fieldStatistics.displayName', {
+  defaultMessage: 'Field statistics',
+});
+
+export const FieldStatisticsInitializer: FC<FieldStatsInitializerProps> = ({
+  initialInput,
+  onCreate,
+  onCancel,
+  onPreview,
+}) => {
+  const {
+    unifiedSearch: {
+      ui: { IndexPatternSelect },
+    },
+    uiSettings,
+  } = useDataVisualizerKibana().services;
+
+  const [dataViewId, setDataViewId] = useState(initialInput?.dataViewId ?? '');
+  const [viewType, setViewType] = useState(
+    initialInput?.viewType ?? FieldStatsInitializerViewType.ESQL
```

Comment:
Is ES|QL the default in Discover yet? (and what about Serverless)? Just wonder if the default should match the default in Discover?